beacon_node/network/src/router/gossip_processor.rs

```diff
@@ -0,0 +1,810 @@
+
+        self.current_workers = self.current_workers.saturating_add(1);
+        let chain = self.beacon_chain.clone();
+        let network_tx = self.network_tx.clone();
+        let sync_tx = self.sync_tx.clone();
+        let log = self.log.clone();
+        let executor = self.executor.clone();
+
+        executor.spawn_blocking(
+            move || {
+                let _worker_timer = worker_timer;
+
+                // We use this closure pattern to avoid using a `return` that prevents the
+                // `idle_tx` message from sending.
+                let handler = || {
+                    match work {
+                        /*
+                         * Unaggregated attestation verification.
+                         */
+                        Work::Attestation(boxed_tuple) => {
+                            let (attestation, subnet_id, should_import) = *boxed_tuple;
+
+                            let _attestation_timer = metrics::start_timer(
+                                &metrics::GOSSIP_PROCESSOR_UNAGGREGATED_ATTESTATION_WORKER_TIME,
+                            );
+                            metrics::inc_counter(
+                                &metrics::GOSSIP_PROCESSOR_UNAGGREGATED_ATTESTATION_VERIFIED_TOTAL,
+                            );
+
+                            let beacon_block_root = attestation.data.beacon_block_root;
+
+                            let attestation = if let Ok(attestation) = chain
+                                .verify_unaggregated_attestation_for_gossip(attestation, subnet_id)
+                                .map_err(|e| {
+                                    handle_attestation_verification_failure(
+                                        &log,
+                                        sync_tx,
+                                        peer_id.clone(),
+                                        beacon_block_root,
+                                        "unaggregated",
+                                        e,
+                                    )
+                                }) {
+                                attestation
+                            } else {
+                                return;
+                            };
+
+                            // Indicate to the `Network` service that this message is valid and can be
+                            // propagated on the gossip network.
+                            propagate_gossip_message(network_tx, message_id, peer_id.clone(), &log);
+
+                            if !should_import {
+                                return;
+                            }
+
+                            metrics::inc_counter(
+                                &metrics::GOSSIP_PROCESSOR_UNAGGREGATED_ATTESTATION_IMPORTED_TOTAL,
+                            );
+
+                            if let Err(e) = chain.apply_attestation_to_fork_choice(&attestation) {
```

Comment:
What about the naive aggregation pool? Don't we also need to add it to that?

Reply:
Great catch!